docs/upgrade-to-v6.md

```diff
@@ -0,0 +1,507 @@
+            requestConfiguration.queryParameters.includeHiddenFolders = "true";
+        });
+```
+
+### Odata Function Parameters
+
+Passing function parameters to requests has changed in v6. The `FunctionOption` class, which extends the `Option` class, has been removed and is no longer used to define function parameters. Futhermore, function and action paramter classes, currently defined by classes with the suffix `ParameterSet`, have been removed. 
+Previously a user would pass function parameters to a request as follows:
+```java
+ReportRootGetMailboxUsageDetailParameterSet parameterSet = ReportRootGetMailboxUsageDetailParameterSet.newBuilder()
+                .withPeriod("<Period Option>") // This is a function parameter
+                .build();
+InputStream usageDetail = graphClient.reports().getMailboxUsageDetail(parameterSet).buildRequest().get();
+```
+In v6 the same call would look like the following:
+```java
+InputStream usageDetail = graphClient.reports().getMailboxUsageDetailWithPeriod("<Period Option>").get();
+```
+
+### Odata Action Parameters
+
+The SDK now supports adding parameters to Odata actions via `RequestBody` classes. This aligns with the removal of `ParameterSet` classes. 
+For example, the `sendMail` action would be called as follows in v5:
+```java
+UserSendMailParameterSet userSendMailParameterSet = UserSendMailParameterSet.newBuilder()
+                .withMessage(new Message())
+                .withSaveToSentItems(true)
+                .build();
+graphClient.me().sendMail(userSendMailParameterSet).buildRequest().post();
+```
+In v6 the same call would look like the following:
+```java
+SendMailPostRequestBody sendMailPostRequestBody = new SendMailPostRequestBody();
+sendMailPostRequestBody.setMessage(new Message());
+sendMailPostRequestBody.setSaveToSentItems(true);
+graphClient.me().sendMail().post(sendMailPostRequestBody);
+```
+
+### Error Handling 
+
+`GraphServiceException` has been removed and is no longer used to handle errors. Instead, the SDK now throws an `ApiException` when an issue occurs with a request. The message of the `ApiException` will contain a more specific error message based on the OdataError returned from the API. `ClientException` remains, however, this is now used to handle less frequent issues with the client itself.
+In v6 a user would handle an error as follows:
+```java
+try {
+    User user = graphClient.me().get();
+} catch (ApiException e) {
+    System.out.println(e.getMessage());
+}
+```
+
+### Drive Item Paths
+
+The current CSDL to OpenAPI conversion process avoids generation of redundant paths, which impacts request builders for driveItems. To mitigate this paths should be available through alternative paths as documented in the reference documentation as seen [here](https://learn.microsoft.com/en-us/graph/api/driveitem-list-children?view=graph-rest-1.0&tabs=http#http-request). 
```

Comment:
```suggestion
The SDK generation process avoids generation of redundant paths, which impacts request builders for driveItems. To mitigate this paths should be available through alternative paths as documented in the reference documentation as seen [here](https://learn.microsoft.com/en-us/graph/api/driveitem-list-children?view=graph-rest-1.0&tabs=http#http-request). 
```

(I believe most people reading this won't have an understanding of our complex pipeline)